src/components/Tabs.react.js

```diff
@@ -0,0 +1,258 @@
+          .tab {
+            width: auto;
+          }
+          .tab--selected {
+            border-bottom: none;
+            border-top: 2px solid ${primaryColor};
+          }
+        }
+      `}</style>
+    </div>
+  );
+};
+
+/**
+ * A Dash component that lets you render pages with tabs - the Tabs component's children
+ * can be dcc.Tab components, which can hold a label that will be displayed as a tab, and can in turn hold
+ * children components that will be that tab's content.
+ */
+export default class Tabs extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      selected: 0
+    };
+
+    this.selectHandler = this.selectHandler.bind(this);
+  }
+  selectHandler(index, value) {
+    this.setState({
+      selected: index
+    });
+    if (this.props.setProps) {
+      this.props.setProps({ value: value });
+    }
+  }
+  render() {
+    let EnhancedTabs;
+
+    if (!Array.isArray(this.props.children)) {
+      // if dcc.Tabs.children contains just one single element, it gets passed as an object
+      // instead of an array - so we put in in array ourselves!
+      this.props.children = [this.props.children];
+    }
+
+    EnhancedTabs = this.props.children.map((child, index) => {
+      // TODO: handle components that are not dcc.Tab components (throw error)
+      // enhance Tab components coming from Dash (as dcc.Tab) with methods needed for handling logic
+      let childProps;
+
+      if (child.props.children) {
+        // if props appears on .children, props are coming from Dash
+        childProps = child.props.children.props;
+      } else {
+        // else props are coming from React (Demo.react.js)
+        childProps = child.props;
+      }
+      return (
+        <EnhancedTab
+          key={index}
+          index={index}
+          id={childProps.id}
+          label={childProps.label}
+          selected={this.state.selected === index}
+          selectHandler={this.selectHandler}
+          className={childProps.className}
+          style={childProps.style}
+          selectedClassName={childProps.selected_className}
+          selected_style={childProps.selected_style}
+          value={childProps.value}
+          mobile_breakpoint={this.props.mobile_breakpoint}
+        />
+      );
+    });
+
+    const tabContainerClass = this.props.vertical
+      ? 'tab-container tab-container--vert'
+      : 'tab-container';
+
+    const tabContentClass = this.props.vertical
+      ? 'tab-content tab-content--vert'
+      : 'tab-content';
+
+    const tabParentClass = this.props.vertical
+      ? 'tab-parent tab-parent--vert'
+      : 'tab-parent';
+
+    return (
+      <div
+        className={`${tabParentClass} ${this.props.parent_className || ''}`}
+        style={this.props.parent_style}
+      >
+        <div
+          className={`${tabContainerClass} ${this.props.className || ''}`}
+          style={this.props.style}
```

Comment:
I had a hard time figuring out what `style` was referring to. Could this be `tab_container_style`? and similarly `tab_container_className`